core/matching/header_matcher.go

```diff
@@ -1,55 +1,60 @@
 package matching
 
 import (
+	"fmt"
 	"strings"
 
+	"github.com/SpectoLabs/hoverfly/core/models"
 	glob "github.com/ryanuber/go-glob"
 )
 
-func CountlessHeaderMatcher(requestMatcherHeaders, toMatch map[string][]string) *FieldMatch {
+func HeaderMatching(requestMatcher models.RequestMatcher, toMatch map[string][]string) *FieldMatch {
 
-	for matcherHeaderKey, matcherHeaderValues := range requestMatcherHeaders {
+	// // Make everything lowercase, as headers are case insensitive
+	// for requestHeaderKey, requestHeaderValues := range toMatch {
+	// 	delete(toMatch, requestHeaderKey)
+	// 	toMatch[strings.ToLower(requestHeaderKey)] = requestHeaderValues
+	// }
+
+	matched := true
+	var matchScore int
 
+	requestMatcherHeadersWithMatchers := requestMatcher.HeadersWithMatchers
+
+	for matcherHeaderKey, matcherHeaderValue := range requestMatcherHeadersWithMatchers {
 		// Make everything lowercase, as headers are case insensitive
 		for requestHeaderKey, requestHeaderValues := range toMatch {
 			delete(toMatch, requestHeaderKey)
 			toMatch[strings.ToLower(requestHeaderKey)] = requestHeaderValues
 		}
+		fmt.Println(toMatch)
+		fmt.Println("matcher key: " + matcherHeaderKey)
+		matcherHeaderValueMatched := false
 
-		toMatchHeaderValues, toMatchHeaderValuesFound := toMatch[strings.ToLower(matcherHeaderKey)]
-		if !toMatchHeaderValuesFound {
-			return FieldMatchWithNoScore(false)
+		toMatchHeaderValues, found := toMatch[strings.ToLower(matcherHeaderKey)]
+		if !found {
```

Comment:
If the header key is not found, do you still need to invoke the `ScoredFieldMatcher`?